app/components/Preview/Types/PreviewUri.tsx

```diff
@@ -12,12 +12,12 @@ export type PreviewUriProps = {
 };
 
 export function PreviewUri(props: PreviewUriProps) {
+  const { isOnline, onceOnline } = useNetworkState();
   const previewFetcher = useFetcher<PreviewResult>();
+  const encodedUri = encodeURIComponent(props.value);
+  const load = () => previewFetcher.load(`/actions/getPreview/${encodedUri}`);
 
-  useEffect(() => {
-    const encodedUri = encodeURIComponent(props.value);
-    previewFetcher.load(`/actions/getPreview/${encodedUri}`);
-  }, [props.value]);
+  isOnline ? load() : onceOnline(load);
```

Comment:
side effects should be called in `useEffect`. Also, this would flood the api with requests if done like this in render function everytime this component rerenders.

```js
  useEffect(() => {
    const encodedUri = encodeURIComponent(props.value);
    const load = () => previewFetcher.load(`/actions/getPreview/${encodedUri}`);
    isOnline ? load() : onceOnline(load);
  }, [isOnline, props.value, onceOnline]);
``` 

but `onceOnline` isn't memoized with a useCallback so, this essentially will trigger on every rerender as well.

A better api would be:

```js
const useLoadWhenOnline = (callback, deps = []) => {
  const callbackRef = useRef<() => void>(callback);

  useEffect(() => {
    callbackRef.current = callback
  })

  useEffect(() => {
        const doCallback = () => callbackRef.current()
        if (window.navigator.onLine) { 
          doCallback();
          return;
        }

        window.addEventListener("online", doCallback);

        return () => {
            window.removeEventListener("online", doCallback);
        };
    }, [...deps]);
}
```

Used as:

```js
useuseLoadWhenOnline(load, encodedUri)
```